labgrid/driver/powerdriver.py

```diff
@@ -232,6 +232,111 @@ def cycle(self):
     def get(self):
         return self.backend.power_get(self._host, self._port, self.port.index)
 
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class NetworkPowerDriverWithAuth(Driver, PowerResetMixin, PowerProtocol):
```

Comment:
Why do you need a further `NetworkPowerDriverWithAuth` function here? How does the netio driver works?
As far as I can remember there is also creds necessary by using the web-api.

Reply:
The netio json-api doesn't need credentials if it's not forced.
We as an user could only use default-credentials or stick the custom credentials to an url.
Why would we parse the Host to extract our user and password, if we could simply provide this as own sub-asset for our resource and process them in our drivers. For example *phoenix_fl_switch* needs a user and password to login via telnet.

Reply:
That might be so for the json-api but if you decide to use the web-api (https://github.com/JUMO-GmbH-Co-KG/labgrid/blob/master/labgrid/driver/power/netio.py#L15) the creds are hardcoded in the driver. 
However, `class NetworkPowerDriverWithAuth(Driver, PowerResetMixin, PowerProtocol):` - your new `class` has the same parameters as the existing `class NetworkPowerDriver(Driver, PowerResetMixin, PowerProtocol):` 
My suggestion would be to use in the first place a given (hardcoded creds like the netio-driver (admin:admin) does), further in a separate MR (step) it might be better to extends just the existing `class NetworkPowerDriver(Driver, PowerResetMixin, PowerProtocol):` with the missing members for `username` and `password`. At the first glance it seems you have a lot of duplicated code by creating your separate `class`.

Reply:
Everything which was introduced with `class NetworkPowerDriverWithAuth` is reverted.
The std-creds were added and it will no longer be possible to set custom credentials. However a future rework with custom creds should be pretty easy.